### PR TITLE
Fix 'Auth fail' error in github-related selenium tests

### PR DESCRIPTION
**Pull request by dmytro-ndp**
Signed-off-by: Dmytro Nochevnov <dnochevnov@codenvy.com>

### What does this PR do?
This request is aimed to avoid failures of github-related selenium tests because of simultaneous run in parallel threads, and problems with clean up of ssh-keys after the tests.

### What issues does this PR fix or reference?
#7018
Pull Request for che 5 https://github.com/eclipse/che/pull/7148

### CODE REVIEW

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java [hunk 1]
```diff
@@ -14,48 +14,95 @@
...
+  private final TestGitHubServiceClient testGitHubServiceClient;
+  private final String gitHubUsername;
+  private final String gitHubPassword;
 
   @Inject
   public TestSshServiceClient(
-      TestApiEndpointUrlProvider apiEndpointProvider, HttpJsonRequestFactory requestFactory) {
+      TestApiEndpointUrlProvider apiEndpointProvider,
+      HttpJsonRequestFactory requestFactory,
+      TestGitHubServiceClient testGitHubServiceClient,
```
Comment (tolusha) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java]: I would have a dedicated service to upload VCS key on github and wouldn't mix ssh and github service

Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java]: Did you mean test service client? Class like the follow:
```
TestGitHubKeyUploader
=========================
TestGitHubServiceClient testGitHubServiceClient
TestSshServiceClient testSshServiceClient
--------------------------------------------
void updateGithubKey()
```

Comment (tolusha) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java]: Something like that.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java [hunk 2]
```diff
@@ -187,6 +192,8 @@ public void deleteAllGrants(final String username, final String password) throws
           .setAuthorizationHeader(createBasicAuthHeader(username, password))
           .request();
     }
+
+    LOG.debug(format("Application grants '%s' were removed from github.com", grandsId));
```
Comment (skabashnyuk) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java]: https://www.slf4j.org/faq.html#logging_performance

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java [hunk 3]
```diff
@@ -14,48 +14,95 @@
...
 import org.eclipse.che.api.ssh.shared.dto.GenerateSshPairRequest;
 import org.eclipse.che.api.ssh.shared.dto.SshPairDto;
 import org.eclipse.che.selenium.core.provider.TestApiEndpointUrlProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** @author Musienko Maxim */
 @Singleton
 public class TestSshServiceClient {
+  private static final Logger LOG = LoggerFactory.getLogger(TestSshServiceClient.class);
```
Comment (skabashnyuk) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java]: use a similar way to get logger in different places. I prefer this one.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java (hunk 1 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java]: okay

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java (hunk 3 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestSshServiceClient.java]: Could you, please, explain, which places did you mean?

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java (hunk 2 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java]: Did you mean usage of parameterized messages?

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/inject/SeleniumTestHandler.java [hunk 4]
```diff
@@ -140,6 +140,9 @@ public void onFinish(ITestContext context) {}
 
   @Override
   public void onStart(ISuite suite) {
+    isCleanUpCompleted.set(false);
+    runningTests.clear();
+
     Runtime.getRuntime().addShutdownHook(new Thread(this::shutdown));
```
Comment (tolusha) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/inject/SeleniumTestHandler.java]: Only one hook should be registered.

Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/inject/SeleniumTestHandler.java]: What do you mean by hook?

**Review by tolusha — APPROVED** [review]

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/inject/SeleniumTestHandler.java (hunk 4 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/inject/SeleniumTestHandler.java]: good catch!

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java (hunk 2 above)
Comment (skabashnyuk) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubServiceClient.java]: yep

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by dmytro-ndp** [comment]
ci-test

**Review by tolusha — APPROVED** [review]

**Pull request merged**